### DISCUSSION TITLE
1.31.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#17852] [General]
# Features
- [#704](https://github.com/Koenkk/zigbee-herdsman/pull/704) Support coordinator mDNS discovery (@Tarik2142)

# Improvements
- [#5706](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5706) Retrieve `occupancy` from Xiaomi RTCGQ15LM interval message (@tyhk)
- [#5722](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5722) Support OTA for ShinaSystem SBM300ZB4, SBM300ZC1, SBM300ZC2, SBM300ZC3, SBM300ZC4 (@nukusinji)
- [#5723](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5723) Support `power_reactive` for Develco SPLZB-131 (@emandtf)
- [#5724](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5724) Support more features for Sinopé thermostats and fix SW2500ZB energy reporting (@ericgu08)
- [#5550](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5550) Improve weekly schedule configuration for Viessman ZK03840 and Ubisys H1 (@sjorge)
- [#5710](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5710) Rework Lidl PSBZS A1 integration (@zinserjan)
- [#5743](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5743) Support `detection_distance` for Xiaomi MCCGQ13LM (@z-x-f)
- [#17662](https://github.com/Koenkk/zigbee2mqtt/issues/17662) Support `energy` and `power` for GE 45857GE (@Koenkk)
- [#5751](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5751) Support `indicator_mode` for Lidl HG06337 (@jlayec)
- [#5752](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5752) Support OTA for Third Reality 3RVS01031Z (@1joyce)
- #16258 Enable Hue effects for Hue brightness only lights (@Koenkk)
- [#5754](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5754) Support `power_on_behavior` for Third Reality 3RSP02028BZ (@weihuan1111)
- [#5757](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5757) Support OTA for Bosch BSP-FZ2 (@DerDreschner)
- [#5770](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5770) Support more features for Xiaomi WS-USC03 (@tmkhoa)
- [#5767](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5767) Enable `linkage_alarm` option by default for Xiaomi `JT-BZ-01AQ/A` and `JY-GZ-01AQ` (@Otnow)
- [#5816](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5816) Expose `filter_age` for IKEA E2007 (@sjorge)
- [#5822](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5822) Expose `x_axis`, `y_axis` and `z_axis` for Third Reality 3RVS01031Z (@1joyce)
- [#5813](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5813) Support `tamper` and `battery` for Develco WISZB-121 (@ultrabug)
- [#15372](https://github.com/Koenkk/zigbee2mqtt/issues/15372) Support `power_on_behavior` and `switch_type` for TuYa TS110E_1gang_1  (@reyko01)
- [#311](https://github.com/Koenkk/zigbee-OTA/pull/311) Support OTA for Develco WISZB-120  (@ultrabug)
- [#5792](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5792) Improvements for Custom devices (DiY) ptvo.info (@ptvoinfo)
- [#17615](https://github.com/Koenkk/zigbee2mqtt/issues/17615) Log TuYa `Unhandled DP` and `NOT RECOGNIZED DP` as debug instead of warn (@Koenkk)
- [#5781](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5781) Support `illuminance_lux ` for Xiaomi ZNCLBL01LM (@desout)
- [#5773](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5773) Support more features for Acova TAFFETAS2 (@molusk)

# Fixes
- [#17607](https://github.com/Koenkk/zigbee2mqtt/issues/17607) Disable unsupported `power_on_behaviour` for LEDVANCE AC26940/AC31266 (@Koenkk)
- [#17594](https://github.com/Koenkk/zigbee2mqtt/issues/17594) Fix incorrect message being published when entity contains `light` or `cover` (@Koenkk)
- #17680 Disable unsupported `power_on_behavior` for Nue / 3A HGZB-01 (@Koenkk)
- #17666 Fix `Cannot use 'in' operator to search for 'power' in undefined` error (@Koenkk)
- [#5750](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5750) Fix unknown power source for Owon WSP403 (@sbeca)
- #16709 Fix no `0` electrical measurements for TuYa TS011F_plug_1 (@vincentscode)
- [#16709](https://github.com/Koenkk/zigbee2mqtt/issues/16709) Fix incorrect `current` value and invalid 0 measurements for TuYa TS011F_plug_1 `_TZ3000_typdpbpg` (@Koenkk)
- [#5734](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5734) Fix `alarm_on_motion` not working for Bosch BWA-1 (@pomeloy)
- [#5720](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5720) Fix incorrect Develco SPLZB-131 `ac_frequency` value (@emandtf)
- [#5711](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5711) Remove vendor name from BMCT-SLZ description (@slugzero)
- [#705](https://github.com/Koenkk/zigbee-herdsman/pull/705) Fix device unnecessary re-interviewed (@slugzero)
- #17527 Fix Lidl 399629_2110 not controllable (@Koenkk)
- #14628 Fix TuYa `_TZ3210_weaqkhab` brightness not controllable (@Koenkk)
- #16709 Fix TuYa TS011F_plug_1  and TS011F_with_threshold not reporting `0` `power`/`current`/`voltage` (@Koenkk)
- [#16460](https://github.com/Koenkk/zigbee2mqtt/issues/16460) Fix Home Assistant `State max length is 255 characters` error for `program` sensor (@Koenkk)
- [#5742](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5742) Fix `Unhandled key` errors for Xiaomi RTCZCGQ11LM (@Otnow)
- [#5746](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5746) Fix Xiaomi RTCGQ13LM OTA update not working (@z-x-f)
- [#17624](https://github.com/Koenkk/zigbee2mqtt/issues/17624) [#17629](https://github.com/Koenkk/zigbee2mqtt/issues/17629) [#17669](https://github.com/Koenkk/zigbee2mqtt/issues/17669) Disable unsupported `power_on_behaviour` for various devices (@koenkk)
- [#17741](https://github.com/Koenkk/zigbee2mqtt/issues/17741) Rename HEIMAN HS3SA to HS3SA/HS1SA (@Koenkk)
- [#17552](https://github.com/Koenkk/zigbee2mqtt/issues/17552) Fix Home Assistant `Payload is not supported (e.g. open, closed, opening, closing, stopped): STOP` error (@Koenkk)
- [#17835](https://github.com/Koenkk/zigbee2mqtt/issues/17835) Fix missing OTA endpoint for Xiaomi SSM-U01 (@Koenkk)
- [#5808](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5808) Fix missing power source of Sonoff SNZB-02D (@octa22)
- [#17364](https://github.com/Koenkk/zigbee2mqtt/issues/17364) Fix occupancy detection not working for TuYa TS0202 (`_TZ3040_bb6xaihh`) (@Koenkk)
- [#17785](https://github.com/Koenkk/zigbee2mqtt/issues/17785) Fix BTicino K4027C/L4027C/N4027C/NT4027C not updating state (@Koenkk)
- [#16709](https://github.com/Koenkk/zigbee2mqtt/issues/16709) Fix `voltage`, `power` and `current` readings of TuYa TS011F_plug (@Koenkk)

# New supported devices
This release adds support for 45 devices: 
- `GW02` Keen Home Signal repeater (@saddukar)
- `b-parasite` Custom devices (DiY) [b-parasite open source soil moisture sensor](https://github.com/rbaron/b-parasite) (@ginkel)
- `1402769` Namron ZigBee LED dimmer (@Koenkk)
- `MC-02` Shade Control Automate shades and blinds with beaded chains (@adarazs)
- `LDSENK02S` ADEO ENKI LEXMAN 16A EU smart plug (@RodriBus)
- `SNZB-01P` SONOFF Wireless button (@liangjia2019)
- `SNZB-02P` SONOFF Temperature and humidity sensor (@liangjia2019)
- `SNZB-04P` SONOFF Contact sensor (@liangjia2019)
- `9135` LED Trading Curtain motor controller (@anharald)
- `S902M-ZG` HZC Motion sensor (@Koenkk)
- `SM0202` TuYa Motion sensor (@Koenkk)
- `T2106` IKEA STOFTMOLN ceiling/wall lamp 15 warm light dimmable (@yschen5812)
- `GL-SD-003P` Gledopto Zigbee DIN Rail triac AC dimmer (@Koenkk)
- `199182` Legrand Wired switch without neutral (@julien-lebot)
- `HSE2936T` Datek Door/window sensor (@cskogstrand)
- `929003128501` Philips Hue Go portable table lamp (@Koenkk)
- `8718801528204` Ynoa Smart LED E27 CCT (@gizmocuz)
- `SM308-2CH` Samotech Zigbee 2 channel in wall switch (@jafrouk)
- `TS0001_switch_1_gang` TuYa 1-Gang switch with backlight (@elad-bar)
- `TS0002_switch_2_gang` TuYa 2-Gang switch with backlight (@elad-bar)
- `TS0003_switch_3_gang` TuYa 3-Gang switch with backlight (@elad-bar)
- `TS0601_switch_4_gang_2` TuYa 4-Gang switch with backlight (@elad-bar)
- `TS0601_smart_human_presence_sensor_2` TuYa Smart Human presence sensor (@Koenkk)
- `TS0601_cover_5` TuYa Curtain/blind switch (@elad-bar)
- `TS0601_cover_with_1_switch` TuYa Curtain/blind switch with 1 Gang switch (@elad-bar)
- `TS0601_cover_with_2_switch` TuYa Curtain/blind switch with 2 Gang switch (@elad-bar)
- `ZNQBKG26LM` Xiaomi Aqara smart wall switch H1M (with neutral, triple rocker) (@Simperdy)
- `800094` Philips Hue Phoenix dimmable LED smart pendant light (@kruton)
- `3RVS01031Z` Third Reality Zigbee vibration sensor (@1joyce)
- `TS0601_cover_6` TuYa Cover motor (@piwko28)
- `HOMA1001_RGBW` Shenzhen Homa Smart LED driver RGBW (@Stefan-3)
- `HOMA1001_RGB` Shenzhen Homa Smart LED driver RGB (@Stefan-3)
- `HOMA1001_CT` Shenzhen Homa Smart LED driver CT (@Stefan-3)
- `HOMA1001_SC` Shenzhen Homa Smart LED driver SC (@Stefan-3)
- `ZLGP14/ZLGP15/ZLGP16` Legrand Wireless and batteryless scenario switch (home arrival/departure, 1-4 switches, daytime day/night) (@Pulsahr)
- `SPM01` TuYa Smart energy monitor for 1P+N system (@tdvtdv)
- `SPLZB-141` Develco Power plug (@westlund)
- `TS0222_temperature_humidity` TuYa Temperature & humidity sensor (@Dualmiracle)
- `TS011F_4` TuYa 2 gang plug (@koenkk)
- `TS0601_switch_8` TuYa 8 Gang switch (@kirovilya)
- `GL-D-009P` Gledopto Zigbee 12W Downlight RGB+CCT (pro) (@mosanwen)
- `GL-D-015P` Gledopto Zigbee 12W Downlight RGB+CCT (pro) (@mosanwen)
- `GL-D-010P` Gledopto Zigbee 12W Downlight RGB+CCT (pro) (@mosanwen)
- `GL-D-013P` Gledopto Zigbee 6W Downlight RGB+CCT (pro) (@mosanwen)
- `SIN-4-RS-20_LEX` NodOn Roller shutter controller (Leroy Merlin version) (@drount)
- `4512763` Namron Zigbee movement sensor (@krikkert)

# Fixed device detections
- #17524 Detect `_TZE200_m9skfctm` as TuYa PA-44Z (@Koenkk)
- #17633 Detect `_TZ3040_bb6xaihh` as TuYa 809WZT (@Koenkk)
- #17610 Detect `_TZE200_cf1sl3tj` as Zemismart ZM85EL-2Z (@Koenkk)
- #17525 Detect `_TZE204_ztc6ggyl` as TuYa ZY-M100-S (@Koenkk)
- #17611 Detect `_TZ3210_alxkwn0h` as TuYa TS0201_1 (@Koenkk)
- #17612 Detect `_TZ3210_jd3z4yig` as TuYa TS0505B_1_1 (@Koenkk)
- #17626 Detect `_TZ3000_h4wnrtck` as Nous E2 (@Koenkk)
- #17526 Detect `_TZ3000_otvn3lne` as TuYa ZMS01 (@Koenkk)
- #17528 Detect `_TZ3000_2putqrmw` as Nous A1Z (@Koenkk)
- [#5739](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5739) Detect `_TZ3000_jwv3cwak` as Nous L12Z (@psi-4ward)
- #17538 Detect `_TZE200_7yoranx2` as Moes ZTRV-ZX-TV01-MS (@Koenkk)
- #17539 Detect `_TZ3000_74hsp7qy` as Zemismart ZN-LC1E (@Koenkk)
- #17537 Detect `_TZ3000_okaz9tjs` as TuYa BSD29 (@Koenkk)
- #17551 Detect `_TZ3000_ruxexjfz` as Nous L13Z (@Koenkk)
- #17679 Detect `_TZ3000_kyb656no` as Meian SW02 (@Koenkk)
- #17560 Detect `_TZ3000_5ucujjts` as Nous ZN231392 (@Koenkk)
- [#5716](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5716) Detect `_TZ3000_a4xycprs` as Moes ZT-SR-EU4 (@kirovilya)
- #17572 Detect `_TZB210_3zfp8mki` as Skydance WZ5_dim_2 (@Koenkk)
- [#17609](https://github.com/Koenkk/zigbee2mqtt/issues/17609) Detect `Plug Value` as LEDVANCE 4058075729322 (@Koenkk)
- [#5708](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5708) Detect `_TZE204_r0jdjrvi` as TuYa TS0601_cover_1 (@darki73)
- [#5714](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5714) Detect `_TZE204_sxm7l9xa` as TuYa TS0601_cover_1 (@richardwonka)
- [#5715](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5715) Detect `_TZ3000_abrsvsou` as TuYa ERS-10TZBVK-AA (@kirovilya)
- [#17586](https://github.com/Koenkk/zigbee2mqtt/issues/17586) Detect `_TZ3000_mdj7kra9` as TuYa DS-111 (@Koenkk)
- [#5725](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5725) Detect `CCT` as Paulmann 50064 (@lukasauer)
- #17549 Detect `_TZE204_6wi2mope` as TuYa TS0601_switch_4_gang (@Koenkk)
- [#5718](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5718) Detect `MOT-C1Z06F` as Profalux NSAV061 (@ralmn)
- [#17590](https://github.com/Koenkk/zigbee2mqtt/issues/17590) Detect `_TZ3000_msl6wxk9` as Aubess 40ZH-O (@Koenkk)
- [#17593](https://github.com/Koenkk/zigbee2mqtt/issues/17593) Detect `_TZ3000_yujkchbz` as Moes MOES_plug (@Koenkk)
- [#5721](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5721) Detect `_TZE204_xsm7l9xa` as TuYa TS0601_smart_human_presence_sensor (@richardwonka)
- [#5733](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5733) Detect `NimlyCode` as Onesti Products AS easyCodeTouch_v1 (@danielkaldheim)
- [#17657](https://github.com/Koenkk/zigbee2mqtt/issues/17657) Detect `_TZ3000_kqvb5akv` as TuYa TS0001_power (@stephdust)
- [#5744](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5744) Detect `_TZE200_ypprdwsl` as TuYa MIR-HE200-TY (@Eales)
- [#17685](https://github.com/Koenkk/zigbee2mqtt/issues/17685) Detect `_TZ3000_k4ej3ww2` as Aubess IH-K665 (@koenkk)
- #17704 Detect `_TZ3000_dziaict4` as TuYa TS0044_1 (@koenkk)
- [#17728](https://github.com/Koenkk/zigbee2mqtt/issues/17728) Detect `_TZ3000_esynmmox` as Lonsonho TS130F_dual (@Koenkk)
- [#17841](https://github.com/Koenkk/zigbee2mqtt/issues/17841) Detect `_TZ3210_dse8ogfy` as Adaprox TS0001_1 (@Koenkk)
- [#5787](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5787) Detect `SV01-412-MP-1.3` as Keen Home SV01 (@saddukar)
- [#17763](https://github.com/Koenkk/zigbee2mqtt/issues/17763) Detect `_TYZB01_sqmd19i1` as TuYa TS0207_water_leak_detector_3 (@Koenkk)
- [#17761](https://github.com/Koenkk/zigbee2mqtt/issues/17761) Detect `_TZ3000_ocjlo4ea` as TuYa TS0207_water_leak_detector_1 (@Koenkk)
- [#17762](https://github.com/Koenkk/zigbee2mqtt/issues/17762) Detect `_TZ3000_upgcbody` as TuYa `TS0207_water_leak_detector_2` (@Koenkk)
- [#17726](https://github.com/Koenkk/zigbee2mqtt/issues/17726) Detect `_TZ3000_wbloefbf` as TuYa TS011F_switch_5_gang (@Koenkk)
- [#17759](https://github.com/Koenkk/zigbee2mqtt/issues/17759) Detect `_TZ3210_it1u8ahz` as TuYa TS0505B_1_2 (@Koenkk)
- [#17637](https://github.com/Koenkk/zigbee2mqtt/issues/17637) Detect `_TZE204_cjbofhxw` as TuYa PJ-MGW1203 (@Koenkk)
- [#17753](https://github.com/Koenkk/zigbee2mqtt/issues/17753) Detect `_TZ3000_l6iqph4f` as Lonsonho TS130F_dual (@Koenkk)
- [#17752](https://github.com/Koenkk/zigbee2mqtt/issues/17752) Detect `_TZ3000_eei0ubpy` as TuYa TS0002_switch_module (@Koenkk)

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/v1.31.0'>1.31.0</a>.</em>

**Comment by BarryS83** [comment]
Upgraded Zigbee2MQTT Home Assistant Addon (HA running on Proxmox) without any issues.

**Comment by Akhundoff** [comment]
Gas Leakage Detector Combustible Sensor . One of the main products of the well-known MOES company. But not supported yet.

https://www.aliexpress.us/item/3256804097660710.html

**Comment by mrJudahBella** [comment]
Auto updated to 1.31.0-1 and crashing. 

Running HA Core 2023.4.6 and HAOS 10.1. 

```
[01:12:28] INFO: Preparing to start...
[01:12:28] INFO: Socat not enabled
[01:12:30] INFO: Starting Zigbee2MQTT...
/app/node_modules/zigbee-herdsman-converters/index.js:79
        if (converter.options) {
                      ^
TypeError: Cannot read properties of undefined (reading 'options')
    at Object.addDefinition [as addDeviceDefinition] (/app/node_modules/zigbee-herdsman-converters/index.js:79:23)
    at new ExternalConverters (/app/lib/extension/externalConverters.ts:15:17)
    at new Controller (/app/lib/controller.ts:84:58)
    at start (/app/index.js:106:18)
```

EDIT: Seems like an external converter issue. I have posted an issue at https://github.com/Koenkk/zigbee2mqtt/issues/17863

**Comment by py3tx** [comment]
Upgraded Zigbee2MQTT Home Assistant Addon (HA running on HassOS) with Tuya TS 0202_1 staying not supported. Made roll back, and everything working ok.

**Comment by zdzieszulice** [comment]
I have the same, a few unsupported sensors, it can be fixed somehow

**Comment by lux73** [comment]
upgrade from 1.30.4 went well - Thx for your hard Work 👍🏻

**Comment by Almighty1985** [comment]
Since updating to 1.31.0 from 1.30.4 every few hours the CPU and RAM usage of the addon goes to 50% and all my devices are "unavaiable". After manually restarting Z2M everything is back to normal for a few hours, until the usage goes up another time. 

The logs looked good (no error message, but also new status update from the devices). Am I the only one with this problem?
